core/modules.rs

```diff
@@ -371,10 +426,16 @@ impl Stream for RecursiveModuleLoad {
           }
           LoadInit::Main(..) | LoadInit::DynamicImport(..) => inner
             .loader
-            .load(inner.op_state.clone(), &module_specifier, None, false)
+            .load(
+              inner.op_state.clone(),
+              &module_specifier,
+              None,
+              inner.is_dynamic_import(),
+            )
             .boxed_local(),
```

Comment:
Oops, the root load here isn't guarded by a module map check.

Reply:
Okay I made some refactors (namely keeping a module map reference in `RecursiveModuleLoad::module_map_rc`) before realising the approach I had in mind didn't work. Went with a [different way](https://github.com/denoland/deno/pull/11089/files#diff-aa01489135c068a2056422f99878ba91861c585baed075470c08c1ac439ab3b2R418-R419) but I want to keep the refactor.

Also removed `LoadInit::MainWithCode(..)` by instead preloading code into the module map when synthetic module code is given.